src/Misc.ino

```diff
@@ -788,24 +788,48 @@ void delayBackground(unsigned long dsdelay)
 }
 
 
+/********************************************************************************************\
+  Check to see if a given argument is a valid taskIndex (argc = 0 => command)
+  \*********************************************************************************************/
+taskIndex_t parseCommandArgumentTaskIndex(const String& string, unsigned int argc)
+{
+  taskIndex_t taskIndex = INVALID_TASK_INDEX;
+  const int ti = parseCommandArgumentInt(string, argc);
+  if (ti > 0) {
+    // Task Index used as argument in commands start at 1.
+    taskIndex = static_cast<taskIndex_t>(ti - 1);
+  }
+  return taskIndex;
+}
+
+
+/********************************************************************************************\
+  Get int from command argument (argc = 0 => command)
+  \*********************************************************************************************/
+int parseCommandArgumentInt(const String& string, unsigned int argc)
+{
+  int value = 0;
+  if (argc > 0) {
+    // No need to check for the command (argc == 0)
+    String TmpStr;
+    if (GetArgv(string.c_str(), TmpStr, argc + 1)) { 
+      value = CalculateParam(TmpStr.c_str()); 
+    }
+  }
+  return value;
+}
+
 /********************************************************************************************\
   Parse a command string to event struct
   \*********************************************************************************************/
 void parseCommandString(struct EventStruct *event, const String& string)
 {
   checkRAM(F("parseCommandString"));
```

Comment:
@TD-er what about to split command string first with special char once, count args and set event->args=count

than all you need to take argument is to find these special chars and cut inside with substring

Reply:
That's a good one.
On the other hand, I am not really sure about the usefulness of this function.
It is only called on a command when following that exact route, but there are more routes to a command.
So I was thinking to parse the commands only where the specific argument is needed.
Also the index definition of what argument number is needed is now a bit of a mess.
Some functions start counting from 0, some from 1 and the arg variable names in the event are also confusing.

So I plan to move to:
- argument index 0 = command
- argument index 1 = first argument => `event->Par1`

Reply:
but parse to find every argument later (lines 828+) takes time, if you can find all you have less work/cpu time needed later

Reply:
so if you will first try to find last argument, you can store/cache all arguments start before it and use this data later

Reply:
I was thinking about adding a `getNextArgv` function, which does use the position and arg count of the previous run.
On top of that it does make sense to keep track of argument positions, but as always it will be a trade-off between CPU cycles and memory.
But I agree that it is not really efficient right now and it will be improved upon. I just didn't want to do too much all at once, since it was already a really big change in code (and improvement in efficiency)